src/main/java/seedu/address/logic/commands/ShowCommand.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.function.Predicate;
+
+import javafx.collections.transformation.FilteredList;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Displays details of the the person in address book whose name matches the argument keyword or is at specified index.
+ * Keyword matching is case insensitive.
+ */
+public class ShowCommand extends Command {
+
+    public static final String COMMAND_WORD = "show";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n1. Displays details of the person with given name, "
+            + "the specified keywords (case-insensitive)\n"
+            + "Parameters: KEYWORD\n"
+            + "Example: " + COMMAND_WORD + " alice\n"
+            + "2. Displays details of the person at the given index\n"
+            + "Parameters : INDEX\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_NO_NAME = "No User with Given Name could be found in the List";
+
+    public static final String MESSAGE_SUCCESS = "Showing details for %1$s.";
+
+    public static final String MESSAGE_MULTIPLE_NAME = "%1$d Persons with the given keyword in their name exists.\n"
+            + "You can try show with index or full name";
+
+    private final Predicate<Person> predicate;
+    private final Index index;
+
+    /**
+     * Constructor for Show Command when name is given
+     *
+     * @param predicate contains Persons with name matching keyword
+     */
+    public ShowCommand(Predicate<Person> predicate) {
+        this.predicate = predicate;
+        this.index = null;
+    }
+
+    /**
+     * Constructor for Show Command when index is given
+     *
+     * @param index of the Person to show
+     */
+    public ShowCommand(Index index) {
+        this.predicate = null;
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
```

Comment:
Nesting is quite deep, and method is very long, making it not very readable. Perhaps some parts could be abstracted out to different functions?